extensions/wikia/TemplateDraft/controllers/TemplateDraftController.class.php

```diff
@@ -58,6 +58,36 @@ public function markTemplateAsNotInfobox() {
 		return true;
 	}
 
+	/**
+	 * Get the converted infobox markup of a given template.
+	 *
+	 * @requestParam string template The name of the template to convert.
+	 */
+	public function getInfoboxMarkup() {
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+
+		$templateName = $this->request->getVal( 'template' );
+		$title = Title::newFromText( $templateName, NS_TEMPLATE );
+		if ( !$title instanceof Title
+			|| !$title->exists()
+			|| !$title->inNamespace( NS_TEMPLATE )
```

Comment:
Parts of it, but for this use case we don't really want or need to check anything to do with the parent or the `templatedraft` permission. IMO there's no reason to restrict just getting the converted wikitext from the API to anything more than `read` which happens automatically for Nirvana controllers, especially as it's aimed at being able to convert a template on one community to be used on another.